drivers/wifi/esp_at/esp.c

```diff
@@ -1008,6 +1009,21 @@ static int esp_mgmt_iface_status(const struct device *dev,
 	return 0;
 }
 
+static int esp_mgmt_mode(const struct device *dev, struct wifi_mode_info *mode)
+{
+	struct esp_data *data = dev->data;
+
+	if (mode->mode == 0 && mode->oper == WIFI_MGMT_SET) {
+		esp_flags_clear(data, EDF_STA_AUTOCONN_ENABLED);
+	}
+
+	if (mode->mode == 1 && mode->oper == WIFI_MGMT_SET) {
+		esp_flags_set(data, EDF_STA_AUTOCONN_ENABLED);
+	}
```

Comment:
`NET_REQUEST_WIFI_MODE` does not look to be designed for "autoconnect" handling. Instead, this API should rather be used for selecting SoftAP, Station, etc.

Reply:
Selecting SoftAP etc is done automatically when `NET_REQUEST_WIFI_AP_ENABLE` or `NET_REQUEST_WIFI_CONNECT` is requested. I want to give the user the option to set autoconnect on/off during runtime. And the only way i could find to communicate with the driver in runtime is via a `NET_REQUEST_WIFI_MODE` request. I have interpeted that `NET_REQUEST_WIFI_MODE` can be made to do whatever you want so thats why i have used it to set/reset autoconnect.

Reply:
I am not sure how to approach this. Overall I agree with you that there might be some cases when autoconnect is desirable on the WiFi chip / firmware side. This is true for ESP-AT or even other "native" ESP32 targets, where there is a networking stack which already implements that functionality.

On the other hand it is more flexible to have control over (auto)connect from Zephyr. However this adds more code on the application side.

I think we could start with an issue describing the problem and potential solution (if possible). Or you can try to modify this implementation, but IMO we need some special purpose API for that, not the existing `NET_REQUEST_WIFI_MODE` which is for other purposes.

Reply:
The `set_mode` was introduced to handle setting of Wi-Fi operations modes like STA, SAP, Promiscuous, Monitor, TX injeection etc. This was needed because for some modes e.g., monitor, mode information was needed before any actions were taken rather than just configuring everything in oneshot like AP_ENABLE or WIFI_CONNECT.

This is also analogous to https://wireless.wiki.kernel.org/en/users/Documentation/iw#adding_interfaces_with_iw .

Auto-connect is a feature flag and not a mode, so, this API isn't suitable. 

Can we add this `auto_connect` to `wifi_connect_params`? This indicates that once the connection is lost, it can keep retrying if this flag isn't set it's just one-shot attempt?

Reply:
> Can we add this auto_connect to wifi_connect_params? This indicates that once the connection is lost, it can keep retrying if this flag isn't set it's just one-shot attempt?

Sounds reasonable.

@krish2718 Curious, do you have a use-case for `autoconnect` in context of chip(s) that you work with?

Reply:
Yes, Nordic's solution uses WPA supplicant which is auto-connect by default unless user issues `disconnect`, so, putting this in the `wifi_connect_params` helps us as some customer didn't want `auto-connect` and needed more control.

Reply:
Great, then we have multiple WiFi solutions with similar feature. Makes sense to go forward with the API update in `wifi_connect_params`.

Reply:
Good, i will implement this into `wifi_connect_params`.